pkgs/applications/graphics/unigine_heaven/default.nix

```diff
@@ -0,0 +1,66 @@
+{ lib,
+  stdenv,
+  fetchurl,
+  makeWrapper,
+  libX11,
+  libXext,
+  libXrandr,
+  freetype,
+  fontconfig,
+  libXrender,
+  libXinerama,
+  autoPatchelfHook,
+  libGL,
+  openal,
+}:
+
+let
+  version = "4.0";
+in
+
+stdenv.mkDerivation {
+  name = "Unigine_Heaven";
+  pname = "Unigine_Heaven";
+
+  src = fetchurl {
+    url = "https://assets.unigine.com/d/Unigine_Heaven-${version}.run";
+    sha256 = "19rndwwxnb9k2nw9h004hyrmr419471s0fp25yzvvc6rkd521c0v";
+  };
+
+  unpackPhase = "true";
+
+  opl = openal; #need for wrapProgram
+  lgl = libGL;
+
+  installPhase = ''
+    cp $src extract.run
+    chmod +x ./extract.run
+    ./extract.run --target $out
+    wrapProgram $out/heaven --prefix LD_LIBRARY_PATH : $libGL/lib:$out/bin:$opl/lib --run "cd $out"
+    mv $out/heaven $out/bin/heaven
```

Comment:
This should be before wrapProgram.

wrapProgram does two things:
* it renames the binary to $out/.heaven-wrapped
* it creates a shell script named 'heaven' that sets the specified environment variables and then executes the .heaven-wrapped binary

It still technically works since the generated shell script uses the full nix store path for .heaven-wrapped, but .heaven-wrapped will be left at the top of $out which is kind of besides the point of having a bin/ directory in the first place.

If you move the 'heaven' binary first, and then call wrapProgram on it, everything will be in bin/. Alternatively you can keep the order like this and additionally move '.heaven-wrapped' to $out/bin/.heaven-wrapped, but moving the original binary first and wrapping it when it is in the right place is cleaner.